lib/apiservers/portlayer/swagger.json

```diff
@@ -175,6 +176,222 @@
+						"description": "failed to export tar archive from target"
+					},
+					"default": {
+						"description": "error",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					}
+				}
+			},
+			"post": {
+				"description": "Imports a tar stream to a device at the target path",
+				"summary": "Import tar stream to supplied target",
+				"tags": [
+					"storage"
+				],
+				"consumes": [
+					"application/x-tar"
+				],
+				"operationId": "ImportArchive",
+				"parameters": [
+					{
+						"name": "ID",
+						"in": "query",
+						"type": "string",
+						"x-nullable": false
+					},
+					{
+						"name": "deviceID",
+						"in": "query",
+						"required": true,
+						"type": "string",
+						"x-nullable": false
+					},
+					{
+						"name": "destination",
+						"in": "query",
+						"required": false,
+						"type": "string",
+						"x-nullable": false
+					},
+					{
+						"name": "filterSpec",
+						"in": "query",
+						"required": false,
+						"type": "string",
+						"x-nullable": true
+					},
+					{
+						"name": "Archive",
+						"required": true,
+						"in": "body",
+						"schema": {
+							"type": "string",
+							"format": "binary"
+						}
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK"
+					},
+					"404": {
+						"description": "Supplied target not found",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					},
+					"409": {
+						"description": "Device or resource has an unresolvable conflict",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					},
+					"500": {
+						"description": "failed to export tar archive from target"
+					},
+					"default": {
+						"description": "error",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					}
+				}
+			},
+			"head": {
+				"description": "Fetches filesystem stats on the device id at the specified path",
+				"operationId": "StatPath",
+				"tags": [
+					"storage"
+				],
+				"consumes": [
+					"application/json"
+				],
+				"produces": [
+					"application/json"
+				],
+				"parameters": [
+					{
+						"name": "store",
+						"in": "query",
+						"type": "string",
+						"x-nullable": false
+					},
+					{
+						"name": "deviceID",
+						"in": "query",
+						"required": true,
+						"type": "string",
+						"x-nullable": false
+					},
+					{
+						"name": "targetPath",
+						"in": "query",
+						"required": true,
+						"type": "string",
+						"x-nullable": false
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"headers": {
+							"name": {
+								"type": "string",
+								"x-nullable": false
+							},
+							"mode": {
+								"type": "integer",
+								"format": "uint32"
+							},
+							"size": {
+								"type": "integer",
+								"format": "int64"
+							},
+							"linkTarget": {
+								"type": "string",
+								"x-nullable": false
+							}
+						}
+					},
+					"404": {
```

Comment:
Since this isn't a streaming handler, you can return our model, Error, and then add more context in the errors returned from the portlayer.  For instance,
```
"404" : {
	"description": "Supplied target not found",
	"schema": {
		"$ref": "#/definitions/Error"
	}
},
```